homeassistant/components/airzone/climate.py

```diff
@@ -105,8 +115,42 @@ def __init__(
         self._attr_hvac_modes = [
             HVAC_MODE_LIB_TO_HASS[mode] for mode in self.get_airzone_value(AZD_MODES)
         ]
+        if (
+            self.get_airzone_value(AZD_SPEED) is not None
+            and self.get_airzone_value(AZD_SPEEDS) is not None
+        ):
+            self._set_fan_speeds()
+
         self._async_update_attrs()
 
+    def _set_fan_speeds(self) -> None:
+        self._attr_supported_features |= ClimateEntityFeature.FAN_MODE
+
+        speeds = self.get_airzone_value(AZD_SPEEDS)
+        max_speed = max(speeds)
+        if max_speed == 2:
+            self._speeds = {
+                0: FAN_AUTO,
+                1: FAN_LOW,
+                2: FAN_HIGH,
+            }
+        elif max_speed == 3:
+            self._speeds = {
+                0: FAN_AUTO,
+                1: FAN_LOW,
+                2: FAN_MEDIUM,
+                3: FAN_HIGH,
+            }
+        else:
+            for speed in speeds:
+                if speed == 0:
+                    self._speeds[speed] = FAN_AUTO
+                else:
+                    self._speeds[speed] = f"{int(round((speed * 100) / max_speed, 0))}%"
+
```

Comment:
Also if you want this to work with HomeKit you have to use the constants FAN_LOW, FAN_MEDIUM, FAN_HIGH, FAN_AUTO, etc.

Someone might ask you for that later ... 😉

Reply:
Its not so great, but mixing probably is the way to currently offer the most functionality

Fan speeds are always hard because you end up with something like:

auto
low
medium
75%
high

I'm tempted to add support for N% to homekit so we can avoid something like that though so maybe best to leave as-is.

Reply:
Done in ef47a1af222b45c548a573555a4aaaea09c336a1